AK/Math.h

```diff
@@ -996,14 +997,18 @@ constexpr T pow(T x, T y)
         return 0;
     if (y == 1)
         return x;
-    int y_as_int = (int)y;
-    if (y == (T)y_as_int) {
-        T result = x;
-        for (int i = 0; i < fabs<T>(y) - 1; ++i)
-            result *= x;
-        if (y < 0)
-            result = 1.0l / result;
-        return result;
+
+    // Take an integer fast path as long as the value fits within a 64-bit integer.
+    if (y >= static_cast<T>(NumericLimits<i64>::min()) && y < static_cast<T>(NumericLimits<i64>::max())) {
```

Comment:
Shouldn't this be `<= NumericLimits<i64>::max()` on the rhs?

Reply:
Nope, that caused issues before.

Reply:
But we use `static_cast<T>(NumericLimits<i64>::min())` on the lhs, and fabs() of that is larger than::max().

What kind of issues, and why don't they happen with the negative limit?

Reply:
okay okay i will let CI decide, please let me be done with this commit already